cluster_management/src/main/java/com/pinterest/rocksplicator/monitoring/mbeans/RocksplicatorSpectatorMonitor.java

```diff
@@ -0,0 +1,78 @@
+package com.pinterest.rocksplicator.monitoring.mbeans;
+
+import org.apache.helix.monitoring.StatCollector;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+public class RocksplicatorSpectatorMonitor
+    implements RocksplicatorSpectatorMonitorMBean {
+
+  public static final String INSTANCENAME_KEY = "InstanceName";
+
+  public enum LATENCY_TYPE {
+    CONFIG_GENERATOR_EXECUTION
+  }
+
+  private final String instanceName;
+  private long configGeneratorCalledCount = 0;
+  private long configGeneratorNullExternalView = 0;
+
+  private ConcurrentHashMap<LATENCY_TYPE, StatCollector> latencyMonitorMap =
+      new ConcurrentHashMap<LATENCY_TYPE, StatCollector>();
+
+
+  public RocksplicatorSpectatorMonitor(String instanceName) {
+    this.instanceName = instanceName;
+
+    latencyMonitorMap.put(LATENCY_TYPE.CONFIG_GENERATOR_EXECUTION, new StatCollector());
+    reset();
+  }
+
+  public void incrementConfigGeneratorCalledCount(int count) {
+    configGeneratorCalledCount += count;
+  }
+
+  void addLatency(LATENCY_TYPE type, double latency) {
+    assert (latencyMonitorMap.containsKey(type));
+    latencyMonitorMap.get(type).addData(latency);
+  }
```

Comment:
Why assert here? if the latency type has not previously been registered... isn't it better to create it ? rather then assert?

Reply:
We initiate/reset all the stats targeted at the Constructor of RocksplicatorSpectatorMonitor, so, later addLatency need to make sure only add stats to those stats targeted.